### PR TITLE
[Bug][DAO] Update db init script and soft_version (#9628)

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of the pull request
* Fix bugs in db init script, as described here:
https://github.com/apache/dolphinscheduler/issues/9628#issuecomment-1104274868
* Update `soft_version`
* Duplicate errors stated in https://github.com/apache/dolphinscheduler/issues/9628#issuecomment-1104274868 caused by `dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema` are not handled in this PR. Therefore, #9628 should NOT be closed until these errors are fixed.

## Brief change log
* Described above

## Verify this pull request
* Compile the project and check output `create_schema.sh` and `soft_version` are as expected.

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/sql/soft_version [hunk 1]
```diff
@@ -1 +1 @@
-2.0.4
+3.0.0
```
Comment (zhongjiajie) [inline dolphinscheduler-dao/src/main/resources/sql/soft_version:1]: Sorry for the reply later. Our version in `pom.xml` use version `dev` but we here use `3.0.0`

**Review by SbloodyS — APPROVED** [review]
LGTM

**Comment by SbloodyS** [comment]
@EricGao888 Would you please add these configure steps to docs? Thanks.

**Comment by EricGao888** [comment]
> @EricGao888 Would you please add these configure steps to docs? Thanks.

@SbloodyS Sure, I will link a new PR to this one: #9598

**Comment by SbloodyS** [comment]
@zhuangchong @caishunfeng @zhongjiajie Guys. Do you have time to take a look at this PR?

**Comment by EricGao888** [comment]
> @EricGao888 Would you please add these configure steps to docs? Thanks.

@SbloodyS BTW, do u have any suggestions for the SQL duplicate key & index errors caused by 2.1.0_schema? Apparently we cannot simply delete it. https://github.com/apache/dolphinscheduler/issues/9628#issuecomment-1104274868

**Review by caishunfeng — APPROVED** [review]
+1

**Comment by EricGao888** [comment]
> +1

@SbloodyS @caishunfeng  Just curious, I think db initialization is a basic and vital operation. Not sure why this bug hasn't been detected by our CI. Should I open an issue to add a check for this in CI?

**Comment by SbloodyS** [comment]
> > +1
> 
> @SbloodyS @caishunfeng Just curious, I think db initialization is a basic and vital operation. Not sure why this bug hasn't been detected by our CI. Should I open an issue to add a check for this in CI?

Currently cluster ```start/stop/create/upgrade``` steps test does not included in our CI. I'm working on it in #9474

File: dolphinscheduler-dao/src/main/resources/sql/soft_version (hunk 1 above)
Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/soft_version:1]: @zhongjiajie Sure I will fix it.

**Comment by EricGao888** [comment]
> > > +1
> > 
> > 
> > @SbloodyS @caishunfeng Just curious, I think db initialization is a basic and vital operation. Not sure why this bug hasn't been detected by our CI. Should I open an issue to add a check for this in CI?
> 
> Currently cluster `start/stop/create/upgrade` steps test does not included in our CI. I'm working on it in #9474

Looks cool! I believe it will help make the code more robust : )

File: dolphinscheduler-dao/src/main/resources/sql/soft_version (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/soft_version:1]: I'm not familiar about dev. If we change ```soft_version``` to ```dev```. Will upgrade sql script be executed or skiped?

Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/soft_version:1]: Actually I have the same question here. I will give a try to check it.

Comment (caishunfeng) [reply dolphinscheduler-dao/src/main/resources/sql/soft_version:1]: This is related to the upgrade script, which will judge the number of the version, so if it is dev, there may be problems.

**Pull request merged**